Polishing changes (#87)
canonical/ubuntu-maintainers-handbook: Concepts.md

> # High level concepts

Author: This page contains "master" terminology, which we should avoid. Is there a preferred alternative?

Reviewer: The following alternatives would work. They are ordered (best to worst) by how much I like them:
1. "*shared*" e.g.:
```diff
- All packages for all Ubuntu releases are stored together in one master directory
+ All packages for all Ubuntu releases are stored together in one shared directory
```
2. "*common*" e.g.:
```diff
- All packages for all Ubuntu releases are stored together in one master directory
+ All packages for all Ubuntu releases are stored together in one common directory
```
3. "*general*" e.g.:
```diff
- All packages for all Ubuntu releases are stored together in one master directory
+ All packages for all Ubuntu releases are stored together in one general directory
```
4. "*main*" e.g.:
```diff
- All packages for all Ubuntu releases are stored together in one master directory
+ All packages for all Ubuntu releases are stored together in one main directory
```
Note: "*main*" is problematic, because it could be confused with the main pocket (see: https://github.com/canonical/ubuntu-maintainers-handbook/blob/main/Concepts.md#main)

An overriding principle in Ubuntu is that processes should make common cases
easy to handle. In exceptional cases, a request for an exception is filed with
the appropriate review team, and if granted, will permit the deviation.

## Ubuntu release cadence

Ubuntu and Debian have distinct differences in infrastructure, procedures, and
release schedules.

Ubuntu follows a strict time-based release cycle. Every six months, a new
Ubuntu version is released and its set of packages declared "stable".
Simultaneously, a new version begins development; it is given its own codename
but also referred to as the "current development release", or "devel".

The development process follows a defined schedule, with periods of active
development followed by various freezes where the type of packaging activity
changes to focus more on stabilisation and preparations for release.

After an Ubuntu version is released, the software packages are altered only for
pressing reasons such as fixing defects or updating critically important
components. These post-release updates to the stable versions of Ubuntu are
referred to as "Stable Release Updates", or "SRUs". SRU work is a primary

> activity of Ubuntu packagers, and will be discussed through this document.

Author: Good catch, thanks!

```text
bash_4.2-2ubuntu2.6.diff.gz         2014-10-09 12:18    90K
bash_4.2-2ubuntu2.6.dsc             2014-10-09 12:18    2.2K
bash_4.2-2ubuntu2.6_amd64.deb       2014-10-09 12:18    626K
bash_4.2-2ubuntu2.6_i386.deb        2014-10-09 12:18    601K
bash_4.2-2ubuntu2.diff.gz           2012-04-03 16:05    79K
bash_4.2-2ubuntu2.dsc               2012-04-03 16:05    1.5K
bash_4.2-2ubuntu2_amd64.deb         2012-04-03 16:05    626K
bash_4.2-2ubuntu2_i386.deb          2012-04-03 16:07    601K
bash_4.2.orig.tar.gz                2011-02-24 01:04    4.1M
bash_4.3-6ubuntu1.debian.tar.gz     2014-04-09 00:58    76K
bash_4.3-6ubuntu1.dsc               2014-04-09 00:58    1.6K
bash_4.3-6ubuntu1_amd64.deb         2014-04-09 01:18    561K
bash_4.3-6ubuntu1_i386.deb          2014-04-09 01:18    535K
bash_4.3-7ubuntu1.7.debian.tar.gz   2017-05-17 17:03    98K
bash_4.3-7ubuntu1.7.dsc             2017-05-17 17:03    2.2K
bash_4.3-7ubuntu1.7_amd64.deb       2017-05-17 17:03    561K
bash_4.3-7ubuntu1.7_i386.deb        2017-05-17 17:03    535K
```

All packages for all Ubuntu releases are stored together in one master
directory, and pointed to by the various releases. This means that you cannot

> have two packages with the same version, even across releases, but you can have

Author: That does help, thanks!

#### Restricted

Software from this repository may not have been tested as extensively as that
contained in the main release, although it includes newer versions of some
applications which may provide useful features. Software in backports will not
receive any review or updates from the Ubuntu security team.

#### Universe

> Software from this repository is entirely unsupported by the Ubuntu team.

Author: Are there any packages from Universe that the Server team supports?

> Software from this repository is entirely unsupported by the Ubuntu team.

Reviewer 2: On a day to day basis we do fix issues and do updates to quite a few universe packages. Officially, on our list of server packages - https://bugs.launchpad.net/~ubuntu-server/+packagebugs - there are a few universe packages listed too, such as python-ws4py, ruby-json, older lua, php, and ruby versions, etc. The divide between main and universe is a bit more arbitrary to us compared to the security team, but in general we do focus on main packages more.

> Software from this repository is entirely unsupported by the Ubuntu team.

Author: Would it be more accurate to say that software from Universe is "mostly unsupported by the Ubuntu Server team"? (since we're really only documenting things from our team's perspective)

> Software from this repository is entirely unsupported by the Ubuntu team.

Reviewer 2: I think that's reasonable, with the caveat that we often support universe packages that are directly related to main server packages

Homepage: https://github.com/ipmitool/ipmitool
```

Here are just a few examples of fields that help us understand where to look
and who to contact about this package:

1. **Maintainer**

   This field typically gives us information on who created the package. If we
   have suggestions or want to propose something, we should talk to the
   maintainer whose email is given in that field.

   Here we see that the maintainer is a person (not a group).

2. **Homepage**

>    Here we can find information about the upstream project's home page URL, and
>    we can find subject matter experts.

Reviewer 2: Reformatting to something like this may work better:

```suggestion
   Here we find the upstream project's home page URL, which may lead us to more
   information and subject matter experts.
```
